README.md

```diff
@@ -492,6 +495,92 @@ status:
+
+    ```yaml
+    schema:
+      openAPIV3Schema:
+        properties:
+          containers:
+            description: Containers describes which containers in a Pod should be bound to
+            items:
+              anyOf:
+              - type: integer
+              - type: string
+              x-kubernetes-int-or-string: true
+            type: array
+    ```
+
+A `CustomBindingPath` resource **MUST** define a `.status.conditions` which is an array of `Condition` objects.  A `Condition` object **MUST** define `type`, `status`, and `lastTransitionTime` entries.  At least one condition containing a `type` of `Ready` **MUST** be defined.  The `status` of the `Ready` condition **MUST** have a value of `True`, `False`, or `Unknown`.  The `lastTransitionTime` **MUST** contain the last time that the condition transitioned from one status to another.  A `CustomBindingPath` resource **MAY** define `reason` and `message` entries to describe the last `status` transition.  There could be multiple application resources matching the `apiVersion` and `kind`.  Partial failures **MUST** be aggregated and reported on the binding status's `Ready` condition.
+
+The reconciler **MUST** set the `CustomBindingPath`'s `Ready` condition according to the same rules set in [Ready Condition Status](#ready-condition-status) for `ServiceBinding` resource.
+
+[cbp-crd]: internal.service.binding_bindingpathmaps.yaml
+[volume]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#volume-v1-core
+[volumemount]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#volumemount-v1-core
+[env]: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#envvar-v1-core
+
+### Resource Type Schema
+
+```yaml
+apiVersion: internal.service.binding/v1alpha2
+kind: CustomBindingPath
+metadata:
+  name:                 # string
+  generation:           # int64, defined by the Kubernetes control plane
+  ...
+spec:
+  application:          # ObjectReference-like
+    apiVersion:         # string
+    kind:               # string
+
+  bindingPaths:
+    volume:             # string, optional
+    volumeMount:        # string, optional, volume must be specified along with this
+    env:                # string, optional
+    containers:         # string, optional
+
+status:
+  conditions:           # []Condition containing at least one entry for `Ready`
+  - type:               # string
+    status:             # string
+    lastTransitionTime: # Time
+    reason:             # string
+    message:            # string
+  observedGeneration:   # int64
+```
+
+### Custom Binding Paths Example
+
+```yaml
+apiVersion: internal.service.binding/v1alpha2
+kind: BindingPathMap
+metadata:
+  name: example-binding-paths
+spec:
+  application:
+    apiVersion: apps.examples.org/v1alpha1
+    kind: ExampleAppBuilder
+
+  bindingPaths:
+    volume: spec.volume
+    volumeMount: spec.volumeMount
+    env: spec.env
+    containers: spec.containers
```

Comment:
@baijum I was trying to say that [corev1.Container](https://godoc.org/k8s.io/api/core/v1#Container) defines both `env` and `volumeMounts` fields. So defining a path to containers is also implicitly defining the path to env and volume mounts. The use of these fields should be mutually exclusive.

Reply:
I think explicitly specifying `volume` and `volumeMount` is better than using `corev1.Container`. In the case of the former, it can be deep copied and inserted in the right container.  In the case of `corev1.Container`, it requires some more processing to extract `volume` and `volumeMount` and then inserting in the right containers.

Reply:
In my mind the cases where a CRD has a volume / volumeMount / env in a different pattern than PodSpec'able, then it most likely won't have the flexibility to pick which container that belongs to.  For example the [RuntimeComponent](https://github.com/application-stacks/runtime-component-operator/blob/master/doc/user-guide.adoc) CRD, which is deploying an app, so things like volumeMount / env belongs specific to the container it creates for that app.

So my vote would be for this PR to keep it simple for now and just specify env / volume / volumeMount mappings.  We can always add more data in the future given more use cases.  If we really have a case where the container is also configurable, then I agree with @scothis about mutual exclusivity with the env / volumeMount fields.